src/uu/nl/src/nl.rs

```diff
@@ -106,6 +106,42 @@ impl NumberFormat {
     }
 }
 
+enum SectionDelimiter {
+    Header,
+    Body,
+    Footer,
+    None,
+}
+
+struct SectionDelimiterParser {
+    delimiter: String,
+}
+
+impl SectionDelimiterParser {
+    fn new(delimiter: &str) -> Self {
+        let delimiter = if delimiter.chars().count() == 1 {
+            format!("{delimiter}:")
+        } else {
+            delimiter.to_owned()
+        };
+
+        Self { delimiter }
+    }
+
+    fn parse(&self, s: &str) -> SectionDelimiter {
+        if self.delimiter.is_empty() {
+            return SectionDelimiter::None;
+        }
```

Comment:
Does making `self.delimiter` an `Option<SectionDelimiter>` make sense? Or maybe the whole `SectionDelimiterParser`?